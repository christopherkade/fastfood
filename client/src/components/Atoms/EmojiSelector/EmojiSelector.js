import React from "react"
import PropTypes from "prop-types"
import styled from "styled-components"

const EmojiButton = styled.button`
  background: rgba(255,255,255, 0.5);
  text-align: center;
  padding: 1.5rem;
  font-size: 1.5rem;
  border: ${props => props.isSelected ? "1px solid red" : "1px solid white"};

  &:hover {
    cursor: pointer;
    opacity: 0.8;
  }

  &:focus {
    outline: 1px solid #EF798A;
  }

  @media (max-width: 1280px) {
    padding: 0.5rem;
  }
`

const EmojiSelector = ({ name, emoji, onClick, isSelected }) => {
  return (
    <EmojiButton onClick={() => onClick(name, emoji)} title={name} isSelected={isSelected}>
      <span role="img" aria-label={name}>{emoji}</span>
    </EmojiButton>
  )
}

EmojiSelector.propTypes = {
  name: PropTypes.string.isRequired,
  emoji: PropTypes.string.isRequired,
  onClick: PropTypes.func.isRequired
}

export default EmojiSelector
