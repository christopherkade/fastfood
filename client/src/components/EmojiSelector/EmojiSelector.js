import React from "react"
import styled from "styled-components"

const EmojiButton = styled.button`
  background: rgba(255,255,255, 0.5);
  text-align: center;
  padding: 1.5rem;
  font-size: 1.5rem;
  border: 1px solid white;

  &:hover {
    cursor: pointer;
    opacity: 0.8;
  }

  &:focus {
    outline: 1px solid #EF798A;
  }
`

const EmojiSelector = ({ name, emoji, onClick }) => {
  return (
    <EmojiButton onClick={() => onClick(name, emoji)}>
      <span role="img" aria-label={name}>{emoji}</span>
    </EmojiButton>
  )
}

export default EmojiSelector
