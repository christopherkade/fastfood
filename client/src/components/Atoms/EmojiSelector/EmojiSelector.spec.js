import React from 'react'
import { render, cleanup, fireEvent } from 'react-testing-library'

import EmojiSelector from "./EmojiSelector"

afterEach(cleanup);

describe("<EmojiSelector />", () => {
  describe("Snapshots", () => {
    it("should match snapshot", () => {
      const { container } = render(<EmojiSelector emoji="üçï" name="Pizza" onClick={() => { }} />);
      expect(container).toMatchSnapshot();
    });
  })

  describe('emoji prop', () => {
    it('should display the emoji passed as a prop', () => {
      const { getByText } = render(<EmojiSelector emoji="üçï" name="Pizza" onClick={() => { }} />);
      expect(getByText("üçï")).toBeTruthy()
    });
  });

  describe('onClick prop', () => {
    it('should call onClick function when Button is clicked', () => {
      const onClickMock = jest.fn();
      const { getByText } = render(<EmojiSelector emoji="üçï" name="Pizza" onClick={onClickMock} />);
      fireEvent.click(getByText('üçï'));
      expect(onClickMock).toHaveBeenCalled();
    });
  });
})
