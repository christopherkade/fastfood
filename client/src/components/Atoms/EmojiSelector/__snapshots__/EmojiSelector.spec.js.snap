// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<EmojiSelector /> Snapshots should match snapshot 1`] = `
<div>
  <button
    class="sc-bdVaJa evWdMy"
    title="Pizza"
  >
    <span
      aria-label="Pizza"
      role="img"
    >
      üçï
    </span>
  </button>
</div>
`;
